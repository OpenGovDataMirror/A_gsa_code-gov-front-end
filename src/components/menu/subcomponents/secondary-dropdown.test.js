import React from 'react'
import { shallow } from 'enzyme'

import SecondaryDropdown from 'components/menu/subcomponents/secondary-dropdown'

const props = {
  menuOption: {
    name: 'option-name',
    links: [
      { name: 'link-1', url: '/link-1-url' },
      { name: 'link-2', url: 'http://link-2-url' },
      { name: 'link-3', url: 'mailto:link-3-url' },
    ],
  },
  onClick: jest.fn(),
}

let wrapper
describe('components - Menu - SecondaryDropdown', () => {
  beforeEach(() => {
    wrapper = shallow(<SecondaryDropdown {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render nothing if no menu option links', () => {
    wrapper.setProps({ menuOption: { ...props.menuOption, links: undefined } })
    expect(wrapper.html()).toBeFalsy()
  })
})