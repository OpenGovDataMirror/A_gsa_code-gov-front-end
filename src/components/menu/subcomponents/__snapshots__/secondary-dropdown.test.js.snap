// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Menu - SecondaryDropdown should render correctly 1`] = `
<ul
  className="usa-nav__submenu"
  hidden={true}
  id="option-name-menu"
>
  <li
    className="usa-nav__submenu-item"
    key="/link-1-url"
  >
    <LinkPart
      name="link-1"
      onClick={[MockFunction]}
      url="/link-1-url"
    />
  </li>
  <li
    className="usa-nav__submenu-item"
    key="http://link-2-url"
  >
    <LinkPart
      name="link-2"
      onClick={[MockFunction]}
      url="http://link-2-url"
    />
  </li>
  <li
    className="usa-nav__submenu-item"
    key="mailto:link-3-url"
  >
    <LinkPart
      name="link-3"
      onClick={[MockFunction]}
      url="mailto:link-3-url"
    />
  </li>
</ul>
`;
